pkg/genericapiserver/server/openapi/openapi.go

```diff
@@ -79,6 +81,15 @@ func (o *openAPI) init(webServices []*restful.WebService) error {
 			return r.Operation, nil, nil
 		}
 	}
+	if o.config.GetDefinitionName == nil {
+		o.config.GetDefinitionName = func(_, name string) (string, []string) {
+			return name[strings.LastIndex(name, "/")+1:], nil
+		}
+	}
+	o.definitions = o.config.GetDefinitions(func(name string) spec.Ref {
+		defName, _ := o.config.GetDefinitionName(o.servePath, name)
+		return spec.MustCreateRef("#/definitions/" + openapi.EscapeJsonPointer(defName))
```

Comment:
There are two tools that are currently using this. The documentation (cc @pwittrock) and python-client. For the python client, with no pre-processing, it will break the client generator and we need to update pre-processing step and re-generate it. The other option is not to use "/" as separator.